src/main/java/net/atos/zac/documenten/InboxDocumentenService.java

```diff
@@ -0,0 +1,122 @@
+import net.atos.zac.documenten.model.InboxDocument;
+import net.atos.zac.shared.model.ListParameters;
+import net.atos.zac.shared.model.SortDirection;
+
+@ApplicationScoped
+@Transactional
+public class InboxDocumentenService {
+
+    @PersistenceContext(unitName = "ZaakafhandelcomponentPU")
+    private EntityManager entityManager;
+
+    @Inject
+    private ZRCClientService zrcClientService;
+
+    @Inject
+    private DRCClientService drcClientService;
+
+    public void delete(final UUID zaakinformatieobjectUUID) {
+        final ZaakInformatieobject zaakInformatieobject = zrcClientService.readZaakinformatieobject(zaakinformatieobjectUUID);
+        final UUID enkelvoudiginformatieobjectUUID = URIUtil.parseUUIDFromResourceURI(zaakInformatieobject.getInformatieobject());
+        final InboxDocument inboxDocument = findByEnkelvoudiginformatieobjectUUID(enkelvoudiginformatieobjectUUID);
+        if (inboxDocument != null) {
+            entityManager.remove(inboxDocument);
+        }
+    }
+
+    public InboxDocument create(final UUID enkelvoudiginformatieobejctUUID) {
+        final EnkelvoudigInformatieobject informatieobject = drcClientService.readEnkelvoudigInformatieobject(enkelvoudiginformatieobejctUUID);
+        final InboxDocument inboxDocument = new InboxDocument();
+        inboxDocument.setEnkelvoudiginformatieobjectID(informatieobject.getIdentificatie());
+        inboxDocument.setEnkelvoudiginformatieobjectUUID(enkelvoudiginformatieobejctUUID);
+        inboxDocument.setCreatiedatum(informatieobject.getCreatiedatum().atStartOfDay(ZoneId.systemDefault()));
+        inboxDocument.setTitel(informatieobject.getTitel());
+        inboxDocument.setBestandsnaam(informatieobject.getBestandsnaam());
+        entityManager.persist(inboxDocument);
+        return inboxDocument;
+    }
+
+    public List<InboxDocument> list(final ListParameters listParameters) {
+        final CriteriaBuilder builder = entityManager.getCriteriaBuilder();
+        final CriteriaQuery<InboxDocument> query = builder.createQuery(InboxDocument.class);
+        final Root<InboxDocument> root = query.from(InboxDocument.class);
+        if (listParameters.getSorting() != null) {
+            if (listParameters.getSorting().getDirection() == SortDirection.ASCENDING) {
+                query.orderBy(builder.asc(root.get(listParameters.getSorting().getField())));
+            } else {
+                query.orderBy(builder.desc(root.get(listParameters.getSorting().getField())));
+            }
+        }
+        final TypedQuery<InboxDocument> emQuery = entityManager.createQuery(query);
+        if (listParameters.getPaging() != null) {
+            emQuery.setFirstResult(listParameters.getPaging().getFirstResult());
+            emQuery.setMaxResults(listParameters.getPaging().getMaxResults());
+        }
+        return emQuery.getResultList();
+    }
+
+    public InboxDocument findByEnkelvoudiginformatieobjectUUID(final UUID enkelvoudiginformatieobjectUUID) {
```

Comment:
Je hebt de 2 delete methods op id en uuis niet een aparte naam gegeven. Dus of de delete(uudi) method een andere naam geven of deze methode gewoon find noemen

Reply:
hernoemd naar find